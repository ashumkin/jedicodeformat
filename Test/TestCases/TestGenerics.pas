unit TestGenerics;


{ AFS November 2007 

  This code compiles, but is not semantically meaningfull.
  It is test cases for the code-formating utility

  Test new generics syntax - code from TridenT
}


interface

uses
  SysUtils;

type
  TGenericType<AnyType> = class
    FData: AnyType;
    function GetData: AnyType;
    procedure SetData(Value: AnyType);
    property Data: AnyType read GetData write SetData;
  end;

implementation

function TGenericType<AnyType>.GetData: AnyType;
begin
  Result := FData;
end;

procedure TGenericType<AnyType>.SetData(Value: AnyType);
begin
  FData := Value;
end;

type
  TGenericTypeInt = TGenericType<Integer>;

var
  I: TGenericTypeInt;

initialization
 try
  With TGenericType<Integer>.Create do
   Data := 100;

  I := TGenericTypeInt.Create;
  I.Data := 100;
  WriteLn(I.Data);

  ReadLn;
 except
  on E:Exception do
   Writeln(E.Classname, ': ', E.Message);
 end;
end.