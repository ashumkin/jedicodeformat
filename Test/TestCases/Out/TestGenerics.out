unit TestGenerics;


{ AFS November 2007 

  This code compiles, but is not semantically meaningfull.
  It is test cases for the code-formating utility

  Test new generics syntax - code from TridenT
}


interface

uses
  SysUtils;

type
  TGenericType<AnyType> = class
    FData: AnyType;
    function GetData: AnyType;
    procedure SetData(Value: AnyType);
    property Data: AnyType Read GetData Write SetData;
  end;

implementation

function TGenericType<AnyType>.GetData: AnyType;
begin
  Result := FData;
end;

procedure TGenericType<AnyType>.SetData(Value: AnyType);
begin
  FData := Value;
end;

type
  TGenericTypeInt = TGenericType<integer>;

var
  I: TGenericTypeInt;

initialization
  try
    with TGenericType<integer>.Create do
      Data := 100;

    I      := TGenericTypeInt.Create;
    I.Data := 100;
    WriteLn(I.Data);

    ReadLn;
  except
    on E: Exception do
      Writeln(E.ClassName, ': ', E.Message);
  end;
end.
